packages/desktop/src/renderer/app/services/environments.service.ts

```diff
@@ -800,9 +804,11 @@ export class EnvironmentsService extends Logger {
       this.store.getDatabucketByUUID(databucketUUID);
 
     if (databucketToDuplicate) {
-      const newDatabucket: DataBucket = this.dataService.renewDatabucketUUIDs(
+      let newDatabucket: DataBucket = this.dataService.renewDatabucketUUIDs(
         CloneObject(databucketToDuplicate)
       );
+      newDatabucket = this.dataService.deduplicateDatabucketID(newDatabucket);
```

Comment:
Maybe we shouldn't renew the ID when copying to another env to avoid breaking ref inside templates